openshift/WFLY-14936_New_S2I_Image.adoc

```diff
@@ -0,0 +1,204 @@
+
+* https://github.com/wildfly/wildfly-s2i[WildFly s2i]
+
+* https://github.com/wildfly/quickstart[WildFly quickstarts]
+
+* https://github.com/shipwright-io/build[Shipwright, S2I build framework]
+
+* ODO dev files.
+
+=== Relevant Installation Types
+// Remove the x next to the relevant field if the feature in question is not relevant
+// to that kind of WildFly installation
+* [ ] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [ ] Bootable jar
+
+== Requirements
+
+=== Implementation notes
+
+In WildFly s2i repo, a new git branch v2 will be used to implement the feature and release the first versions of the images. The master will be used to 
+build and release the current WildFly S2I and runtime images that will become deprecated. The build and release of the legacy cloud Galleon feature-pack 
+is also done in the master branch.
+Once the deprecated images are no more released, a legacy branch will be created, the v2 branch will be merged into master. At this point, no more "legacy" content will be released.
+
+=== Hard Requirements
+
+==== WildFly s2i builder image
+
+===== Building
+
+* The builder image is built from ubi8-minimal image.
+* It adds on top JDK11 and maven s2i tooling.
+* It contains the user jboss.
+* It is deployed on quay.io: ```quay.io/wildfly/wildfly-s2i-jdk11```
+* Has a versioning independent from WildFly versioning
+* Has support for binary and source build.
+* Expects a server to be provisioned in ```<target dir>/server``` by default. Directory name relative to the target directory can be configured with 
+ ```S2I_SERVER_DIR=<relative path in target directory>``` env variable.
+* The image should be usable using https://github.com/shipwright-io/build[Shipwright build framework] on both Kubernetes and Openshift.
+
+
+===== Running
+
+* Image produced with this builder image is runnable.
+* Image execution can be configured with Java related environment variables. https://github.com/jboss-openshift/cct_module/tree/master/jboss/container/java/jvm/api[JVM API]
+* The image entry-point offers the following features:
+** Discover the WildFly launcher script to be called: ```standalone.sh, standalone-cloud.sh or openshift-launch.sh```. 
+N.B.: In case of openshift-launch.sh (legacy entry point), no configuration is applied. Configuration of the server is delegated 
+to the ```openshift-launch.sh``` script that handles it all.
+** Computes the JVM options to be conveyed to the server
+** Handles POD termination with CLI shutdown. Can be disabled with ```CLI_GRACEFUL_SHUTDOWN=true``` env variable.
+** Set default value for management (0.0.0.0) and public (value of ```hostname -i``` command) interfaces. 
+    The default values can be specified with ```SERVER_PUBLIC_BIND_ADDRESS``` and  ```SERVER_MANAGEMENT_BIND_ADDRESS```.
+** Enables the server statistics. Can be overridden with ```SERVER_ENABLE_STATISTICS=false```.
```

Comment:
once https://issues.redhat.com/browse/WFCORE-5464 is merged, we will be able to do that "natively" in WildFly using the `WILDFLY_STATISTICS_ENABLED` env var instead

Reply:
That is a good point. Do we want the statistics enabled by default when running in the image? So have -Dwildfly.statistics-enabled=[true by default but overridden by SERVER_ENABLE_STATISTICS] passed in all cases?
If that is not the case, we can remove this "default" and let the user uses the SERVER_ARGS env variable to provide it explicitly (explicitly SERVER_ARGS=-Dwildfly.statistics-enabled=true) or use WFCORE-5464.
I had the feeling that we wanted it by default in cloud context.